### PR TITLE
[LoongArch64] add cross supporting LoongArch64.

### PR DESCRIPTION
**Pull request by shushanhf**
add cross supporting LoongArch64.

### CODE REVIEW

**Comment by shushanhf** [comment]
Hi, @am11 
Could you review this PR please ?

**Comment by am11** [comment]
@shushanhf, how was this tested? As discussed previously, unless we can provide corresponding support in build-rootfs.sh, adding code for new architecture in toolchain.cmake alone does not make much sense because configuration in this file depends on the exact setup done by rootfs script.

**Comment by shushanhf** [comment]
> @shushanhf, how was this tested? As discussed previously, unless we can provide corresponding support in build-rootfs.sh, adding code for new architecture in toolchain.cmake alone does not make much sense because configuration in this file depends on the exact setup done by rootfs script.

OK, Thanks.
I will upload it after the corresponding upstream supports LoongArch64.

**Comment by am11** [comment]
For reference, riscv64 support was initially added in llvm9 (current version is llvm13) and we are still waiting for dependencies like lldb. Therefore, I haven't pushed that change yet. LoongArch64 support is not even available in vanilla llvm13 toolchain, so I think we can wait at least until patches are merged in its mainline. Then look to port other dependencies (krb5, openssl, lttng-ust, lldb, omp5 etc.) along with our preferred debootstrap tool used on Debian, before changing rootfs script.

TL;DR, there is no rush for upstreaming cross-build setup right now.

**Comment by shushanhf** [comment]
> For reference, riscv64 support was initially added in llvm9 (current version is llvm13) and we are still waiting for dependencies like lldb. Therefore, I haven't pushed that change yet. LoongArch64 support is not even available in vanilla llvm13 toolchain, so I think we can wait at least until patches are merged in its mainline. Then look to port other dependencies (krb5, openssl, lttng-ust, lldb, omp5 etc.) along with our preferred debootstrap tool used on Debian, before changing rootfs script.

Thanks for your advices.
Are qemu and docker's image also needed ?

Now all these can only be getted throuth the pulic patches or mirrors which are still be pushing to upstream.

**Comment by am11** [comment]
> Now all these can only be getted throuth the pulic patches or mirrors which are still be pushing to upstream.

Unreviewed / unmerged patches are something you can use in your local setup for the initial bootstrap. You can create your own docker container with prerequisites to build dotnet-runtime and share it with people who are interested, but unless dependencies get loongarch64 support upstream, it is not something we should push in arcade; the .NET engineering infrastructure (I think rust engineering infra has the similar requirements for new platforms).

**Comment by akoeplinger** [comment]
Superseded by https://github.com/dotnet/arcade/pull/15370
